File: python/organizing_policy/get_game.py
from organizing_policy.game_id import GameId
from organizing_policy.game_repository import GameRepository


def make_get_game(game_repository: GameRepository):
    def get_game(game_id: GameId):
        game = game_repository.find_by_id(game_id)
        if game is None:
            raise ValueError("game not found")
        return game

    return get_game

File: ruby/organizing_policy/lib/organizing/get_game.rb
require_relative 'game_id'
require_relative 'game_repository'

module Organizing
  class GetGame
    def initialize(game_repository)
      @game_repository = game_repository
    end

    def call(game_id)
      game = @game_repository.find_by_id(game_id)
      raise ArgumentError, 'Game not found' if game.nil?
      game
    end
  end
end
